doc/classes/RenderingServer.xml

```diff
@@ -838,6 +838,8 @@
 		<method name="create_local_rendering_device" qualifiers="const">
 			<return type="RenderingDevice" />
 			<description>
+				Creates a RenderingDevice to do all local drawing and compute on any thread. Cannot draw to the screen nor share data with the global RenderingDevice.
+				[b]Note:[/b] When using the OpenGL backend or when running in headless mode, this function always returns [code]null[/code].
```

Comment:
I wonder if this is intentional for headless, or just that the implementation is missing? I could imagine users wanting to run headless on a system with a GPU to do compute stuff.

Still, if that's the current state the documentation is correct, we should just open an issue if we think that headless RenderingDevice ought to be supported.

Reply:
`--headless` does not have a way to use the GPU – this would require implementing off-screen rendering: https://github.com/godotengine/godot-proposals/issues/5790

Reply:
Do you think there would be interest in a feature to make a compute-only local rendering device, while the rest of the engine is headless? There would be no window created and no off-screen rendering.

I have a toy example capable of running the [compute demo](https://github.com/dzil123/godot-demo-projects/tree/compute_shader_headless/misc/compute_shader_heightmap) on Linux without a display: https://github.com/godotengine/godot/compare/master...dzil123:godot:headless_local_renderingdevice.

Reply:
Yeah I think that would make sense. This implementation would need more work of course as there shouldn't be X11 includes in `servers`, platform-specific code should stay self-contained within in platform folder.

Reply:
> platform-specific code should stay self-contained within in platform folder

Right, of course. The specific implementation difficulty for that is the need to get the platform specific `VulkanContext` impl. Currently only the platform specific `DisplayServer`s reference the specific VulkanContext, but I can't use the DisplayServer singleton to construct the VulkanContext because on headless the singleton is `DisplayServerHeadless`. This may need a mechanism like `DisplayServer::create_func`.

There is also the question of whether creating a VulkanContext should be allowed if using the OpenGL renderer, and in general how this feature should be controlled through project settings or the CLI.